app/src/main/java/ru/vssemikoz/newsfeed/data/LocalNewsStorage.java

```diff
@@ -18,18 +19,18 @@ public LocalNewsStorage(NewsAppDataBase dataBase) {
     }
 
     @Override
-    public List<NewsItem> getFiltered(Boolean favoriteNewsState, Category category) {
+    public Single<List<NewsItem>> getFiltered(Boolean favoriteNewsState, Category category) {
```

Comment:
You should return a single `Single` from the method. Something like:
`Single.fromCallable(new Callable<List<NewsItem>() {`
`    override List<NewsItem> call() {`
`        // your method body`
`    }`
`});`